src/main/java/com/zendesk/maxwell/util/ListWithDiskBuffer.java

```diff
@@ -19,33 +19,26 @@
 	private ObjectInputStream is;
 	private ObjectOutputStream os;
 
-	public ListWithDiskBuffer(long maxInMemoryElements) throws IOException {
+	public ListWithDiskBuffer(long maxInMemoryElements) {
 		this.maxInMemoryElements = maxInMemoryElements;
 		list = new LinkedList<>();
 	}
 
 	public void add(T element) throws IOException {
 		list.add(element);
 
-		while (this.list.size() > maxInMemoryElements) {
-			if ( file == null ) {
-				file = File.createTempFile("maxwell", "events");
-				file.deleteOnExit();
-				os = new ObjectOutputStream(new BufferedOutputStream(new FileOutputStream(file)));
-			}
-
-			if ( elementsInFile == 0 )
-				LOGGER.debug("Overflowed in-memory buffer, spilling over into " + file);
-
-			os.writeObject(this.list.removeFirst());
-
-			elementsInFile++;
-
-			if ( elementsInFile % maxInMemoryElements == 0 )
-				os.reset(); // flush ObjectOutputStream caches
-		}
+		while ( shouldBuffer() )
+			evict();
 	}
-
+//*
```

Comment:
oh I see the override for `shouldBuffer`, guess just get rid of the comment markers and `resetOutputStreamCaches` unless there is some reason to keep it